lxd/instance_post.go

```diff
@@ -199,13 +179,74 @@ func instancePost(d *Daemon, r *http.Request) response.Response {
 			return resp
 		}
 	} else if sourceNodeOffline {
-		// If a target was specified, forward the request to the relevant node.
+		// If a target was specified, forward the request to the relevant node/member.
 		resp := forwardedResponseIfTargetIsRemote(s, r)
 		if resp != nil {
 			return resp
 		}
 	}
 
+	inst, err := instance.LoadByProjectAndName(s, projectName, name)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// Run the cluster placement after potentially forwarding the request to another member.
+	if target != "" && clustered {
+		err = s.DB.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+			_, targetGroup := shared.TargetDetect(target)
+
+			p, err := dbCluster.GetProject(ctx, tx.Tx(), projectName)
+			if err != nil {
+				return err
+			}
+
+			targetProject, err = p.ToAPI(ctx, tx.Tx())
+			if err != nil {
+				return err
+			}
+
+			allMembers, err := tx.GetNodes(ctx)
+			if err != nil {
+				return fmt.Errorf("Failed getting cluster members: %w", err)
+			}
+
+			targetMemberInfo, err = project.CheckTarget(ctx, r, tx, targetProject, target, allMembers)
+			if err != nil {
+				return err
+			}
+
+			if targetMemberInfo == nil {
+				clusterGroupsAllowed := project.GetRestrictedClusterGroups(targetProject)
+
+				candidateMembers, err = tx.GetCandidateMembers(ctx, allMembers, []int{inst.Architecture()}, targetGroup, clusterGroupsAllowed, s.GlobalConfig.OfflineThreshold())
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		})
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		// If no member was selected yet, pick the member with the least number of instances.
+		if targetMemberInfo == nil {
+			err := s.DB.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+				targetMemberInfo, err = tx.GetNodeWithLeastInstances(ctx, candidateMembers)
+				return err
+			})
+			if err != nil {
+				return response.SmartError(err)
+			}
+		}
+
+		if targetMemberInfo.IsOffline(s.GlobalConfig.OfflineThreshold()) {
+			return response.BadRequest(fmt.Errorf("Target node is offline"))
```

Comment:
This is a user facing error that mentions "node", but it should say "cluster member".

Reply:
Missed this one, fixed.